src/signals/incident/containers/IncidentContainer/reducer.js

```diff
@@ -33,29 +33,34 @@ export const initialState = fromJS({
     },
     questions: [],
     source: undefined,
-    subcategory: '',
     type: {
       id: 'SIG',
       label: 'Melding',
     },
   },
   loadingClassification: false,
   usePredictions: true,
-  subcategoryPrediction: '',
+  categoryPrediction: null,
 });
 
-const getIncidentWithoutExtraProps = (incident, { category, subcategory } = {}) => {
-  const prevCategory = incident.get('category');
-  const prevSubcategory = incident.get('subcategory');
-  const hasChanged = prevCategory !== category || prevSubcategory !== subcategory;
+const getIncidentWithoutExtraProps = (incident, category) => {
+  const prevCategory = incident.get('category')?.toJS();
+  const hasChanged = prevCategory?.slug !== category?.slug;
 
-  if (!hasChanged && category && subcategory) return incident;
+  if (!hasChanged && category?.slug) return incident;
 
   const seq = Seq(incident).filter((val, key) => !key.startsWith('extra'));
 
   return seq.toMap();
 };
 
+const getIncidentCategory = (state, category) => {
+  const prevCategory = state.get('incident').get('category')?.toJS();
+  const categoryPrediction = state.get('categoryPrediction')?.toJS();
+  const canChange = categoryPrediction === null || prevCategory === null || prevCategory?.slug === categoryPrediction?.slug;
+  return canChange ? category : prevCategory;
```

Comment:
Checking the coverage, this branch isn't covered.

Reply:
Good catch! Thanks!